tests/test_engage.py

```diff
@@ -99,6 +99,21 @@ def test_git_query(git_repo, capsys, workspace, monkeypatch):
     with pytest.raises(TypeError):
         git_query()
 
+def test_git_query_new_repo(git_repo_no_commits, capsys, workspace, monkeypatch):
+
+    # Check that git_query correctly determines that the repo is not currently clean
+    assert git_query(git_repo_no_commits, "catalogue_results") == False
+
+    # Select "no" when asked whether to commit local changes; repo will still not be clean
+    monkeypatch.setattr('builtins.input', lambda: "n")
+    assert git_query(git_repo_no_commits, "catalogue_results", True) == False
```

Comment:
Same as above: `assert not git_query(git_repo_no_commits, "catalogue_results", True)`